libs/wingc/src/type_check.rs

```diff
@@ -628,11 +586,16 @@ impl<'a> TypeChecker<'a> {
 
 				// Lookup the type in the env
 				let type_ = self.resolve_type(class, env, statement_idx);
-				let (class_env, class_symbol) = match type_.into() {
-					&Type::Class(ref class) => (&class.env, &class.name),
-					&Type::Resource(ref class) => (&class.env, &class.name), // TODO: don't allow resource instantiation inflight
-					&Type::Anything => return Some(self.types.anything()),
-					_ => panic!("Expected {} to be a resource or class type but it's a {}", class, type_),
+				let (class_env, class_symbol) = match *type_ {
+					Type::Class(ref class) => (&class.env, &class.name),
+					Type::Resource(ref class) => (&class.env, &class.name), // TODO: don't allow resource instantiation inflight
```

Comment:
This TODO is a oneliner right? Just need to check the current env and add an error if it's inflight

Reply:
I believe you're right.

Reply:
done.